src/main/java/net/citizensnpcs/EventListen.java

```diff
@@ -559,4 +570,45 @@ public int hashCode() {
             return prime * (prime * (prime + ((worldName == null) ? 0 : worldName.hashCode())) + x) + z;
         }
     }
+
+    private class SkinUpdateTracker {
+        float initialYaw;
+        final Location location = new Location(null, 0, 0, 0);
+        boolean hasMoved;
+        int rotationCount;
+
+        SkinUpdateTracker(Player player) {
+            reset(player);
+        }
+
+        boolean shouldUpdate(Player player) {
+
+            if (!hasMoved) {
+                hasMoved = true;
+                return true;
+            }
+
+            Location currentLoc = player.getLocation(YAW_LOCATION);
+            float currentYaw = currentLoc.getYaw();
+
+            boolean hasRotated = Math.abs(currentYaw - this.initialYaw) < 90;
```

Comment:
Yes, I understand the turning issue because that's why skins don't load most of the time (I verified in MC source a while ago). I think this is where you get most of the benefits.